filecoin-proofs/src/api/util.rs

```diff
@@ -38,3 +45,27 @@ pub fn get_base_tree_size<Tree: MerkleTreeTrait>(sector_size: SectorSize) -> Res
 pub fn get_base_tree_leafs<Tree: MerkleTreeTrait>(base_tree_size: usize) -> Result<usize> {
     get_merkle_tree_leafs(base_tree_size, Tree::Arity::to_usize())
 }
+
+// Returns a range of nodes read from a sector data/replica file stored at `path`.
+pub fn read_nodes_from_file<D: Domain>(path: &Path, node_range: Range<usize>) -> Result<Vec<D>> {
+    let file = OpenOptions::new()
+        .read(true)
+        .open(path)
+        .with_context(|| format!("failed to open file: {}", path.display()))?;
+
+    let byte_offset = node_range.start * NODE_SIZE;
+    let byte_len = (node_range.end - node_range.start) * NODE_SIZE;
+    let bytes = unsafe {
+        MmapOptions::new()
+            .offset(byte_offset as u64)
+            .len(byte_len)
+            .map(&file)
+            .with_context(|| format!("failed to mmap range of file: {}", path.display()))?
+    };
```

Comment:
Oh, that probably makes supporting data over pipes.. hard

I guess APIs which operate on buffers are also fine, we can manage data reading on Go side - passing large enough buffers over Cgo should make the overhead small enough to be mostly ignored.